rasa/shared/utils/io.py

```diff
@@ -626,3 +628,21 @@ def is_subdirectory(path: Text, potential_parent_directory: Text) -> bool:
 def random_string(length: int) -> Text:
     """Returns a random string of given length."""
     return "".join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+
+def handle_print_blocking(output: Text) -> None:
+    """Handle print blocking (BlockingIOError).
+    Only available to Unix systems since it uses fcntl
+    to directly manipulate I/O control on STDOUT.
+
+    Args:
+        output: Text to be printed to STDOUT.
+    """
+    old = fcntl.fcntl(sys.stdout.fileno(), fcntl.F_GETFL)
```

Comment:
@Urkem This solution doesn't work with Windows (which is why all the CI checks on Windows fail).
I made a note of this in the initial PR [here](https://github.com/RasaHQ/rasa/pull/11934#issuecomment-1446134640), but the Jira ticket was missing this information.

Reply:
Also we need this PR to target `3.5.x`, because it's a bugfix (only features, improvements and maintenance works target main). Please rebase or open a new PR.

Reply:
Thanks for the info will close and create a new PR for 3.5.x, will try using [portalocker](https://github.com/WoLpH/portalocker) and dropping fcntl as well